### DISCUSSION TITLE
Maybe CommitOutboundQueueMessages should fail-fast when shutdown happens in Azure Stroage provider?

### DISCUSSION THREAD

**Discussion by jjshao-ms** [Azure/azure-functions-durable-extension#2443] [Ideas]
Using Azure Storage provider.
I am encountering a host shutdown during my orchestration is running `CommitOutboundQueueMessages`. I notice that even if the ownership lease has been released due to shutdown, the framework is still enqueueing message to the work item queue. It's ok since I know activity might be executed more than one time, but it would be better if we could reduce the number of duplicate executions.

FYI, related [code snippet](https://github.com/Azure/durabletask/blob/53091265df19f18346b94465cc9ef95ffe8f4248/src/DurableTask.AzureStorage/AzureStorageOrchestrationService.cs#L1251-L1253)

**Comment by cgillum** [comment]
Hi @jjshao-ms, it sounds like you want to propose a behavior change. Can you open an issue for this? That will be more effective than a discussion.

**Reply by jjshao-ms** [reply]
Sure.

**Comment by jjshao-ms** [comment]
https://github.com/Azure/durabletask/issues/885
